app/services/invoices/customer_usage_service.rb

```diff
@@ -171,6 +173,26 @@ def compute_amounts_with_provider_taxes
       invoice.total_amount_cents = invoice.fees_amount_cents + invoice.taxes_amount_cents
     end
 
+    # Progressively billed invoices ae created in the subscription boundaries, but because they
+    # do not have paid_in_advance charges, we can not substitute them in the usage (as paid_in_advance_fees).
+    # We need to add the credits that were progressively billed within the same boundaries
+    def add_progressively_billed_usage
```

Comment:
I think you should use the `Subscriptions::ProgressiveBilledAmount` service.

(I think the logic here is wrong when there's 2 progressive invoices in the past month, e.g it should only be the last one taken into account. But the aforementioned service takes care of everything. )